vcd/resource_vcd_snat.go

```diff
@@ -164,16 +187,77 @@ func resourceVcdSNATDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
 	}
 
-	task, err := edgeGateway.RemoveNATMapping("SNAT", d.Get("internal_ip").(string),
-		d.Get("external_ip").(string),
-		"")
+	if "" != d.Get("network_name").(string) {
+		err = edgeGateway.RemoveNATRule(d.Id())
+		if err != nil {
+			return fmt.Errorf("error deleting SNAT rule: %#v", err)
+		}
+	} else {
+		// this for back compatibility when network name and network type isn't provided - TODO remove with major release
+		task, err := edgeGateway.RemoveNATMapping("SNAT", d.Get("internal_ip").(string),
+			d.Get("external_ip").(string),
+			"")
+		if err != nil {
+			return fmt.Errorf("error deleting SNAT rule: %#v", err)
+		}
+		err = task.WaitTaskCompletion()
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func resourceVcdSNATUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	vcdClient := meta.(*VCDClient)
+
+	// Update supports only when network name and network type provided
+	networkName := d.Get("network_name")
+	if nil == networkName || networkName.(string) == "" {
+		return fmt.Errorf("update works only when network_name and network_type is provided and rule created using them \n")
```

Comment:
This is a wider team discussion but we'll have to think how to make it less hard to migrate for that major version 3.0. At least imports could help or could we handle somehow internally and populate the variables based on the data so that user has migration/update path.
Not intending to solve it here, just flagging.
CC @dataclouder, @lvirbalas